internal/featuretests/v3/featuretests.go

```diff
@@ -478,15 +482,10 @@ type Response struct {
 func (r *Response) Equals(want *envoy_service_discovery_v3.DiscoveryResponse) *Contour {
 	r.Helper()
 
+	sort.Slice(want.Resources, func(i, j int) bool { return string(want.Resources[i].Value) < string(want.Resources[j].Value) })
+	sort.Slice(r.Resources, func(i, j int) bool { return string(r.Resources[i].Value) < string(r.Resources[j].Value) })
```

Comment:
This was the biggest change required to get most feature tests to pass, and worth some discussion. It seems that with the go-control-plane server, the order that resources are in when creating the snapshot is not preserved, because internally the slice is converted to a map. Historically, I believe there may have been some inefficiencies in Envoy when resources (particularly Listeners?) were provided in a different order across responses, which is why Contour generally sorted resources before responding with them, but it's definitely possible that those are no longer issues.

Reply:
yeah I think everything besides the route matches in a RouteConfig should be stored in envoy in a constant-time lookup container

but then as you say i see this old issue: https://github.com/projectcontour/contour/issues/1159

i wonder if that is still the case...

Reply:
I'll do a few more tests here and look at the Envoy debug logs to see if I can discern anything, but I have to think if there were any issues related to this, other users of go-control-plane would've run into them by now

Reply:
Seems like envoy computes a hash of each resource and uses that to detect when there are changes. Everything in the logs look fine as far as I can tell.